src/python/qequlacs/utils.py

```diff
@@ -38,24 +73,45 @@ def append_gate(gate, qulacs_circuit):
     elif len(gate.qubits) == 2:
         index_1 = gate.qubits[0].index
         index_2 = gate.qubits[1].index
+
         if gate.name == 'CNOT':
             qulacs_circuit.add_CNOT_gate(index_1, index_2)
         elif gate.name == 'SWAP':
             qulacs_circuit.add_SWAP_gate(index_1, index_2)
         elif gate.name == 'XX':
-            qulacs_circuit.add_multi_Pauli_rotation_gate([index_1, index_2], [1,1], -gate.params[0]*2/np.pi)
+            qulacs_circuit.add_multi_Pauli_rotation_gate([index_1, index_2], [1,1], -angle)
         elif gate.name == 'YY':
-            qulacs_circuit.add_multi_Pauli_rotation_gate([index_1, index_2], [2,2], -gate.params[0]*2/np.pi)
+            qulacs_circuit.add_multi_Pauli_rotation_gate([index_1, index_2], [2,2], -angle)
         elif gate.name == 'ZZ':
-            qulacs_circuit.add_multi_Pauli_rotation_gate([index_1, index_2], [3,3], -gate.params[0]*2/np.pi)
+            qulacs_circuit.add_multi_Pauli_rotation_gate([index_1, index_2], [3,3], -angle)
         elif gate.name == 'CPHASE':
-            mat = np.diag([1., np.exp(1.j*np.pi*gate.params[0])])
-            gate = qulacs.gate.DenseMatrix(index_2, mat)
-            gate.add_control_qubit(index_1,1)
+            matrix = np.diag([1., np.exp(1.j * angle)])
```

Comment:
Again, why are we dropping the additional factor of `np.pi`?

Reply:
@simonwa7  please see my logical path in an answer above,
hope it is a right way